erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -409,17 +409,38 @@ def pr_required(self):
 				return
 
 			for d in self.get("items"):
-				if not d.purchase_receipt and d.item_code in stock_items:
-					msg = _("Purchase Receipt Required for item {}").format(frappe.bold(d.item_code))
-					msg += "<br><br>"
-					msg += _(
-						"To submit the invoice without purchase receipt please set {0} as {1} in {2}"
-					).format(
-						frappe.bold(_("Purchase Receipt Required")),
-						frappe.bold("No"),
-						get_link_to_form("Buying Settings", "Buying Settings", "Buying Settings"),
-					)
-					throw(msg, title=_("Mandatory Purchase Receipt"))
+				received_qty = frappe.db.get_value(
+					"Purchase Order Item",
+					{
+						"name": d.po_detail,
+						"parent": d.purchase_order,
+						"item_code": d.item_code
+					},
+					"received_qty"
+				)
+				invoiced_qty = frappe.get_all("Purchase Invoice Item",
+					{
+						"po_detail": d.po_detail,
+						"item_code": d.item_code,
+						"docstatus": 1
+					},
+					"sum(qty) as qty")[0].get("qty") or 0
```

Comment:
Is there a possibility that this return an empty array? Because if it is possible, then your `list[0]` element access will fail and error

Reply:
I tested without any invoice being created. It didn't failed.